source/FAST/Algorithms/ImagePatch/PatchStitcher.cpp

```diff
@@ -116,7 +116,7 @@ void PatchStitcher::processImage(std::shared_ptr<Image> patch) {
 			m_outputImage = Image::New();
             m_outputImage->create(fullWidth, fullHeight, fullDepth, patch->getDataType(), patch->getNrOfChannels());
         } else {
-            if(fullWidth < 8192 && fullHeight < 8192) {
+            if(fullWidth < 1024 && fullHeight < 1024) {  // lowered the limit from 8192 to 1024
```

Comment:
I think this is too low, and may cause issues for your other WSI pipelines. Like patch-wise classification. If the resulting image of patch-wise classification is higher than 1024x1024, then that pipeline will fail for the same reason as it failed for "high"-res segmentation with 8192x8192 resolution. 
Also, creating an image pyramid for a 1024x1024 image is very overkill.

I think a better solution to your problem is to merge the SegmentationRenderer and the SegmentationPyramidRenderer so that it accepts both Image and ImagePyramid data objects.

Reply:
I agree that merging them would be the ideal solution, but that is outside the scope of what I think I would be able to do with the time I have available. Seems like a lot of work and directly affects all those that are regularly using SegmentationRenderer. Perhaps this is something you could do when you get time? 

One of our end-users need to be able to run inference on image plane = 2, for the images he is using, and there it seems that 8192x8192 is too large. I believe the resultant image plane he was using was somewhat in the order of 2k x 4k, just below the 2048 mark for the WSI I was testing on.